mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergSerDe.java

```diff
@@ -176,12 +179,22 @@ private static Schema hiveSchemaOrThrow(Properties serDeProperties, Exception pr
       if (columnComments != null) {
         Collections.addAll(comments, columnComments.split(Character.toString(Character.MIN_VALUE)));
       }
+      String identifierFields = serDeProperties.getProperty(InputFormatConfig.IDENTIFIER_FIELD_NAMES);
+      Set<String> identifierFieldNames = getIdentifierFieldSet(identifierFields);
       Schema hiveSchema = HiveSchemaUtil.convert(names, TypeInfoUtils.getTypeInfosFromTypeString(columnTypes),
-              comments, autoConversion);
+              comments, autoConversion, identifierFieldNames);
       LOG.info("Using hive schema {}", SchemaParser.toJson(hiveSchema));
       return hiveSchema;
     } else {
       throw new SerDeException("Please provide an existing table or a valid schema", previousException);
     }
   }
+
+  public static Set<String> getIdentifierFieldSet(@Nullable String identifierFields) {
```

Comment:
Could we create a general method here instead, which encapsulates getting the list from the map too? Something like:
```
public static Set<String> getIdentifierFieldSet(Map properties) {
}
```